### ISSUE TITLE
Unconnected Nodes (as displayed in ARAX UI, not displayed at all in Translator UI)

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/Feedback#134]
Steps to Recreate:

1. UI CI, What Drugs May Treat, use "Cole Carpenter Syndrome" or "Takayasu arteritis"
2. View results in Translator UI - looks fine, find and expand Dexamethasone

- CC Syndrome:  https://ui.ci.transltr.io/results?q=db6e146f-78f0-4bb6-aa43-3ac97225650f

- Find and expand Dexamethasone

- TA: https://ui.ci.transltr.io/results?q=c5002e23-c011-49f8-b74f-dc19f4bd1779
- View the individual ARA KGs to see the unconnected nodes https://arax.ncats.io/?r=c5002e23-c011-49f8-b74f-dc19f4bd1779
- for TA this includes, ARAX, BTE, Improve and Unsecret

3. View Results in ARAX UI

- CC Syndrome:  https://arax.ncats.io/?r=db6e146f-78f0-4bb6-aa43-3ac97225650f

4.  Open ARAX results, expand Dexamethasone 
![image](https://user-images.githubusercontent.com/38321826/220967239-abf29b8e-ba34-4c1a-9d1e-b5e090c22b04.png)

Note the ARAX KG representation also has many nodes unconnected
![image](https://user-images.githubusercontent.com/38321826/220967317-1930fec0-4de2-4214-9057-196f6dafd440.png)

6. Open BTE Results, can't find Dexamethasone
7. Search for UMLS C0011777 (which i think is a synonym for dexamethasone based on reviewing BTE's raw json) - it's there, connected and is the source of the results displayed in the UI
![image](https://user-images.githubusercontent.com/38321826/220967598-7f9ae328-b1d1-4dfd-b5c2-fb6fc4a0c68a.png)

8. Check the BTE KG representation, there are other nodes that are node connected
![image](https://user-images.githubusercontent.com/38321826/220967715-d7f2c708-383c-44e6-854f-6a1b4723e7f6.png)

9.  Unsecret -individual results don't load, error 598 but 40 results.  KG has an unconnected node, and it has dexamethasone through a path that is not in the UI
![image](https://user-images.githubusercontent.com/38321826/220968089-da3bce95-20cc-4785-8ed3-dd693fdea201.png)

**Comment by sstemann** [comment]
adding an example from today's standup here which also shows that ImProve has unconnected nodes (At least through the ARAX UI and therefore these results are excluded from the UI)

UI:  https://ui.ci.transltr.io/results?l=Fuchs%27%20Endothelial%20Dystrophy&t=0&q=30fa40cd-e74d-4532-ada1-9914c87db20a
ARAX UI:  https://arax.ncats.io/?r=30fa40cd-e74d-4532-ada1-9914c87db20a

![image](https://user-images.githubusercontent.com/38321826/221970064-976f64d0-e0fc-4d8b-a7be-f626f9536b33.png)

![image](https://user-images.githubusercontent.com/38321826/221970131-5ba9d4e3-8cd0-4794-be76-2816cb751a94.png)

**Comment by dkoslicki** [comment]
Work on this issue is happening here: https://github.com/RTXteam/RTX/issues/1968. Preliminary results indicate that something is getting mangled between the ARS and the ARAX UI

**Comment by edeutsch** [comment]
It appears that the ARS is changing the node identifiers without changing the subject and object fields correspondingly. For example, the native ARAX result has:
```
        "CHEMBL.COMPOUND:CHEMBL71595": {
          "attributes": [
            {
              "attribute_source": null,
              "attribute_type_id": "biolink:IriType",
              "attributes": null,
              "description": null,
              "original_attribute_name": null,
              "value": "https://identifiers.org/chembl.compound:CHEMBL71595",
              "value_type_id": "metatype:Uri",
              "value_url": "https://identifiers.org/chembl.compound:CHEMBL71595"
            },
```

but the ARS version has:
```
        "PUBCHEM.COMPOUND:784": {
          "attributes": [
            {
              "attribute_source": null,
              "attribute_type_id": "biolink:IriType",
              "attributes": null,
              "description": null,
              "original_attribute_name": null,
              "value": "https://identifiers.org/chembl.compound:CHEMBL71595",
              "value_type_id": "metatype:Uri",
              "value_url": "https://identifiers.org/chembl.compound:CHEMBL71595"
            },
```

which would normally okay, except that the ARS version also has edges like:
```
        "infores:rtx-kg2:CHEMBL.COMPOUND:CHEMBL71595-biolink:entity_positively_regulates_entity-UniProtKB:P10909": {
          "attributes": [
            {
              "attribute_source": "infores:rtx-kg2",
              "attribute_type_id": "biolink:aggregator_knowledge_source",
              "attributes": null,
              "description": null,
              "original_attribute_name": null,
              "value": "infores:rtx-kg2",
              "value_type_id": "biolink:InformationResource",
              "value_url": null
            }
          ],
          "object": "UniProtKB:P10909",
          "predicate": "biolink:entity_positively_regulates_entity",
          "qualifiers": null,
          "subject": "CHEMBL.COMPOUND:CHEMBL71595"
```

where the subject still refers to the node id before the change. If the ARS is going to change node CURIEs in the spirit of "normalization", it also needs update other parts of the document that refer to those node CURIEs as well, such as in the edges.

**Comment by dkoslicki** [comment]
From Eric:

>I suspect they have implemented "node normalization" at the ARS. So they are changing our CHEMBL identifiers to PUBCHEM identifiers. But only partially. And ARS is therefore emitting corrupted documents. I don't think this is an ARAX or ARAX GUI issue, but rather an ARS bug that they need to fix.

This appears to be an ARS issue @MarkDWilliams . Glancing at output results (stuff that goes through ARAX vs through the ARS) supports this

**Comment by MarkDWilliams** [comment]
I see.  Yes, this does appear to be an ARS issue in the normalization process.  The subject and object portion should be easy enough to resolve, but I'm not immediately sure how to deal with the URIs in attributes.  Is there some way of programmatically distinguishing these from other attributes with the same value and attribute type ids?

**Comment by edeutsch** [comment]
I don't quite understand the question. Although I will agree that the attribute_type_id in this case is not very good.

**Comment by MarkDWilliams** [comment]
`        "CHEMBL.COMPOUND:CHEMBL71595": {
          "attributes": [
            {
              "attribute_source": null,
              "attribute_type_id": "biolink:IriType",
              "attributes": null,
              "description": null,
              "original_attribute_name": null,
              "value": "https://identifiers.org/chembl.compound:CHEMBL71595",
              "value_type_id": "metatype:Uri",
              "value_url": "https://identifiers.org/chembl.compound:CHEMBL71595"
            },`
 It may just be my misunderstanding.  What I'm seeing is this, there is an attribute attached to this node which contains a link to an identifier for it, but there is no original_attribute_name or other way to tell what this attribute is (and thus how to pick it out of the others as needing to be changed when we change the node CURIE).  Or is it the case that only this attribute will ever have a type of biolink:IriType?

**Comment by edeutsch** [comment]
yeah, I think this is a malformed attribute. I think all it is trying to say is that this is the expansion of the CURIE. But I would say that:
1) This is not desirable to have here because it is universally understood that CHEMBL.COMPOUND: means "https://identifiers.org/chembl.compound:"
2) "biolink:IriType" seems like a weak value for the concept of "The fully expanded URI of this concept is.."

It would be useful to receipt a definitive ruling on this by the Data Modeling Council.

But in any case, to first order, it doesn't matter that much. The crucial part is that subject and object get adjusted. The attributes is less crucial. I wouldn't worry about that part until we get an official ruling on the attribute as a whole.

**Comment by MarkDWilliams** [comment]
Understood.  @ShervinAbd92 and I will handle the issue with the subject/object mismatching in the ARS normalization.

**Comment by sierra-moxon** [comment]
The biolink property I would use for this is `biolink:xref`.

**Comment by ShervinAbd92** [comment]
Hi folks, I pushed a code update on the CI env, that takes care of edge id normalization alongside with node id normalization. please give it a try and let me know if the issue still exists?

**Comment by sstemann** [comment]
this is resolved in CI and the change was pushed to Test and Prod as well @sierra-moxon i suggest closing this ticket
